api/processors/actions/group/discard.py

```diff
@@ -0,0 +1,62 @@
+from django.db import transaction
+
+from api.models import Group, Teacher
+from api.models.auxil.status_setter import StatusSetter
+from api.models.choices.log_event_type import StudentLogEventType, TeacherLogEventType
+from api.models.choices.status import StudentProjectStatus, TeacherProjectStatus
+from api.processors.actions.group import GroupActionProcessor
+from api.processors.auxil.log_event_creator import GroupLogEventCreator
+
+
+class GroupDiscardProcessor(GroupActionProcessor):
+    def __init__(self, group: Group, reason: str):
+        self.reason = reason
+        super().__init__(group)
+
+    @transaction.atomic
+    def process(self) -> None:
+        self._set_statuses()
+        self._create_log_events()
+        self._delete()
+
+    def _create_log_events(self) -> None:
+        GroupLogEventCreator.create(
+            group=self.group,
+            student_log_event_type=StudentLogEventType.TENTATIVE_GROUP_DISCARDED,
+            teacher_log_event_type=TeacherLogEventType.TENTATIVE_GROUP_DISCARDED,
+            comment=self.reason,
+        )
+
+    def _delete(self) -> None:
+        self.group.delete()
+
+    def _set_group_status(self) -> None:
+        super()._set_group_status()
+
+    def _set_coordinators_status(self) -> None:
+        StatusSetter.update_statuses_of_active_coordinators(self.timestamp)
+
+    def _set_teachers_status(self) -> None:
+        teachers = Teacher.objects.filter_active()
```

Comment:
I'm also not sure which statuses should be set to which teachers, need power of @lemontree210 here

Reply:
After reading issue, I thinks here status changes should only be applied to group's teachers, but I'm not sure. Also not sure about `abort` :D

Reply:
> I'm also not sure which statuses should be set to which teachers, need power of @lemontree210 here

what power? are you kidding me 😆

Reply:
> After reading issue, I think here status changes should only be applied to group's teachers, but I'm not sure. Also not sure about `abort` :D

Exactly, here we don't need to change coordinators' statuses because this is just a tentative group and it has no coordinator yet.

`abort` is different because this is already a real group that is listed in coordinator's groups. `abort` removes this group from the set, which warrants recalculation of this coordinator's status.

Reply:
I think the question was about whether to apply statuses to only group's teachers or to the whole set of teachers and the same for abort

Reply:
Oh, OK! Well actually one should only apply statuses to this group's teachers, but we had a discussion with @Roxe322 (I can't remember the exact PR where it was, sorry) and decided that it would be easier to apply the status rules to all teachers. Now, I might be dangerously wrong here, but I don't think it would not lead to unpredictable results (because we're using synchronous Django). And, anyway, a teacher can teach multiple groups and if both groups get aborted, we're getting approximately the same situation.

That being said, yes, the 100% approach would be to only limit the status setters to this group's teachers (and, in case of `abort`, this group's coordinator(s)).

Reply:
In that case, here I should filter that if teacher has more groups then his status is working and he has only this group then set status to no group yet, right?